joomla/jissues
# Use template CDN modules

#### Summary of Changes

The core `joomla.org` template has had some updates to be able to translate and render the top menu and footer elements through our CDN.  So, let's use it.

This also drags in the `joomla/cache` package.  We don't have anything to handle caching data right now, and the Joomla sites have been known to come to a screeching halt when there's a high amount of traffic hitting the CDN node, so we're going to cache the response we get from the CDN nodes for these template elements.

The cache store right now only supports filesystem storage for persistent storage; there's also an in-memory store (for the lifetime of the process) and a no-op adapter if desired but generally for production we won't use those.  You can also customize the cache path for data going through this store, otherwise it defaults to `cache/general` (to not collide with our existing g11n and Twig caches).

The elements are also cached language aware.

This will also improve translator workflow a bit.  Now these template elements will only be translated once and apply globally to all projects using the CDN elements.

#### Testing Instructions

With the patch applied, the menu and footer should both render correctly.  The menu should not include the CMS' `<jdoc>` tag for the search position there and the footer should have the 4 placeholders (copyright year, link for "report an issue", and the log in/out text/link) correctly set.

I believe all of the languages the tracker supports are also included in the [template Crowdin project](https://crowdin.com/project/joomla-official-websites-templates), but in case there is one that isn't, the CDN renderer falls back to English.  So it will always render text.
